pkg/build/builder/common.go

```diff
@@ -266,7 +266,7 @@ func HandleBuildStatusUpdate(build *buildapiv1.Build, client buildclientv1.Build
 		latestBuild.Status.Output.To = build.Status.Output.To
 		latestBuild.Status.Stages = timing.AppendStageAndStepInfo(latestBuild.Status.Stages, build.Status.Stages)
 
-		_, err = client.UpdateDetails(latestBuild.Name, latestBuild)
+		_, err = client.UpdateDetails(build.Name, latestBuild)
```

Comment:
Observed on CI:

```
error: Unable to update build status: resource name may not be empty
```

It's possible that the `latestBuild` object returned is an empty struct and no error is reported. This message is likely masking other errors that would have occurred (image pull timeouts were also observed in the same issue).